legit.py

```diff
@@ -0,0 +1,53 @@
+from git import Repo
+from git.exc import GitCommandError, InvalidGitRepositoryError
+from typing import Iterable
+
+# parent_commits: Tuple[str, str] = (), author: Tuple[str, str] = author,
+# committer: Tuple[str, str] = committer
+# TODO Consider author and commiter param
+# debug and rewrite clone method and others
+# author = Actor("An author", "katridi@yandex.ru")
+# committer = Actor("A committer", "katridi@yandex.ru")
+
+
+class Repository:
+
+    def __init__(self, path: str) -> None:
+        self.repo = Repo.init(path)
+
+    @classmethod
+    def clone(cls, url: str, path: str) -> str:
+        try:
```

Comment:
This method is called clone, it should clone, full stop. If you want to make another factory to create a Repository instance from a working dir, make another method. It should be up to the user of this module to decide if they want one or the other. Or they want the logic you implemented here, which is really unobvious.

Reply:
I've tried to stick with this recommendation https://github.com/sphericalpm/lektorium/pull/18#discussion_r305814938
Maybe I've poorly understood or implemented this method

Reply:
You have done what @apatrushev requested correctly. My comment is about what the method does, not about what it's called